### ISSUE TITLE
Highcharts' boost module takes too long to render chart with string values on xAxis

### ISSUE THREAD

**Issue by alirezameftahi71** [highcharts/highcharts#8222]
#### Expected behaviour
With the same amount of data, I expect highcharts' boost module to be just equally fast with string values as it is with numeric values. 

#### Actual behaviour
With numeric values on xAxis, rendering is fast. 
But with string values on xAxis, rendering is very slow.

#### Live demo with steps to reproduce
We have the exact same amount of data (70K and 2 series) in two charts.
Below is a chart with numeric values on xAxis. You can see this renders less than a second. (481.20ms)
https://jsfiddle.net/alinux7/5mvafayn/ 

And Below is a chart with string values on xAxis. This one takes almost 8 secs. (7623.05ms)
https://jsfiddle.net/alinux7/jzthkvhd/

#### The actual problem
When data is just 70K it's not really a pain, but when I try to add 500K, data in it, it takes more than one minute! 

#### Product version
tested on 6.0.4 & 6.1.0

#### Affected browser(s)
tested on Chrome

**Comment by KacperMadej** [comment]
Hi @alirezameftahi71 

Thank you for reporting about the problem.

The suggestion might go against previous workarounds in related issues #6926 and #6964, but it looks like using data only and not setting the categories in an axis works faster. Demo: https://jsfiddle.net/alinux7/5mvafayn/

Please try to create a demo with data generated or loaded from a file - JSFiddle crashed few times for me when trying to work with your demos because of how big the input for JS was.

**Comment by KacperMadej** [comment]
My bad - I copied wrong demo in my previous comment.
I wanted to share this one: http://jsfiddle.net/BlackLabel/2qnh248b/1/

500k points with categories under 2 seconds. The difference is that I'm not setting categories in axis options, but I'm using point's name values.

**Comment by KacperMadej** [comment]
Additionally, for date and time please use `datetime` axis type to avoid problems with categories - API reference: https://api.highcharts.com/highcharts/xAxis.type

**Comment by alirezameftahi71** [comment]
Thank you so much. This would work great but it has just one problem.
I enabled zooming by :
 `chart: {
      zoomType: 'x'
  }`
When I zoom-in, the labels on xAxis won't show correctly. Is there a fix for this?
Demo here : http://jsfiddle.net/alinux7/2qnh248b/2/

And about the `datetime` type, you are right. That would work great for dateTime values. But this is a mock data for testing purposes.

**Comment by pawelfus** [comment]
Hi @alirezameftahi71 

Could you tell us more about your use case? It's the first time I see someone who wants to display 500k of categories. In general, categories are used with small number of points (< 100) - otherwise categories are really crowded and chart is unreadable. Even zooming is a bit problematic - I can not zoom into specific categories, because I simply don't see them..

**Comment by behnam7171** [comment]
Hi,
I have exactly same issue with xAxis with category when using boost module. @pawelfus is there another way to use string in xAxis and not using the xAxis.categories? because I guess it is normal using string in xAxis with or without using xAxis.categories

**Comment by pawelfus** [comment]
Hi @behnam7171 

You are right, it's possible to use `linear` xAxis with formatter: https://jsfiddle.net/BlackLabel/pjwdrgzx/

**Comment by behnam7171** [comment]
@pawelfus yes. that was nice trick. I guess thats because xAxis.categories replaces all 500k strings in the array in xAxis but formatter does it for the xAxis that are meant to be rendered in UI.  Am I right?

**Comment by pawelfus** [comment]
That's correct, when using categories, Highcharts lib is trying to render every string/category, then hide the overlapping ones. When we use linear axis, we know extremes (e.g. 10 000 - 510 000) and we can calculate the scale before rendering ticks. On a liner axis we don't render every tick (10 001, 10 002 etc.) but every nth-tick according to [`xAxis.tickPixelInterval`](https://api.highcharts.com/highcharts/xAxis.tickPixelInterval) option.
